### PR TITLE
JS-Capstone-2 mealsDB API

### PR DESCRIPTION
**Pull request by iamsjunaid**
For the JavaScript Capstone project, we chose an API called [MealsDB](https://www.themealdb.com/api.php). From the API, we particularly selected those that contain information about Seafood. The web app consists of three interfaces

`The home page.`
`The comments popup.`
`The reservation popup.`

**For the Home page, when the page loads, the web app retrieves data from:**
- The Meals DB API shows the list of meals on the screen.
- When the user clicks on the Like button, the interaction is recorded in the Involvement API and the screen is updated.
- The Comments popup appears when the user clicks on the Comments button.

**For the Comment popup, when the popup loads, the web app retrieves data from:**
- The Meals DB API shows details about the selected item.
- The Involvement API shows the item comments.
- When the user clicks on the Comment button, the data is recorded in the Involvement API and the screen

**For the Reservations popup, when the popup loads, the web app retrieves data from:**
- The Meals DB API shows details about the selected item.
- The Involvement API shows the item reservations.
- When the user clicks on the reserve button, the data is recorded in the Involvement API and the screen.

## 🎥For video walkthrough:

https://github.com/iamsjunaid/js-capstone-2/assets/61676854/28e7a6e6-bd82-4755-acdc-4d15df554039

## 🌍Live demo link 

[live demo](https://iamsjunaid.github.io/js-capstone-2/)

### CODE REVIEW

File: src/modules/getReservations.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    },
+  );
+  const responseJSON = await response.json();
+  return responseJSON;
+};
+
+const getReservations = async (itemId) => {
+  const response = await fetch(
+    `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/tN0rUHEjucNdKTNwHcwx/reservations?item_id=${itemId}`,
+  );
```
Comment (Owusu-Desmond) [inline src/modules/getReservations.js:24]: - [ ] Kindly note that after I refresh the page and click on the reservation button to see the reservation I added, there is no reservation showing so make sure that you fix this to make your project work perfectly. See the screenshot below :point_down: 
  
  image

File: src/modules/tests/postComments.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    comment: 'Hello',
+  };
+
+  const response = await postComments(
+    data.item_id,
+    data.username,
+    data.comment,
+  );
+
+  expect(response).toBe('Created');
```
Comment (Owusu-Desmond) [inline src/modules/tests/postComments.test.js:16]: - [ ] Please you have one failing test so make sure that you fix it to ensure that the app is working perfectly :pray: 
  
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @iamsjunaid, @Peter-Dumbari and @uno36  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the setup of the project ✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/tests/postComments.test.js (hunk 2 above)
Comment (m05e5) [inline src/modules/tests/postComments.test.js:16]: - [ ] Great job fixing the other changes, but you still have one failing test so make sure that you fix it and ensure that the app is working perfectly
image

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iamsjunaid** [comment]
The app is perfectly fine on our computers and all the tests cases are passing perfectly fine. I am attaching a screenshot of test cases done on my computer. Please go through it once again.

Screenshot (270)
Screenshot (269)

File: src/modules/tests/postComments.test.js (hunk 2 above)
Comment (iamsjunaid) [reply src/modules/tests/postComments.test.js:16]: The app is perfectly fine on our computers and all the tests cases are passing perfectly fine. I am attaching a screenshot of test cases done on my computer. Please go through it once again.

Screenshot (270)
Screenshot (269)

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @iamsjunaid, @Peter-Dumbari and @uno36  

To highlight 👍🏽 
✔️ App runs smoothly

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**